### ISSUE TITLE
Thin line between content and footer when using footer-html

### ISSUE THREAD

**Issue by kavishatalsania** [wkhtmltopdf/wkhtmltopdf#4665]
I am generating pdf with `wkhtmltopdf  --margin-left "0mm" --margin-right "0mm" --no-outline --page-size "letter" --header-html "path" --footer-html "path"`

I am getting a thin line between content and footer as shown in image.

![image](https://user-images.githubusercontent.com/32358748/79335798-3f9b5380-7f40-11ea-98c3-92fb1007678e.png)

I've tried `--footer-spacing 0` still not work.

Thank you for your help.

**Comment by PhilterPaper** [comment]
All I can think of at this point is that you have a `<div>` defining background color, and maybe the "box model" is a little bit different between your browser and WebKit. This would be whether margin and border count as within the width and height of the div. Anyway, that's something that you might be able to pursue.

**Comment by kavishatalsania** [comment]
I think this is because of footer only as I've tried generating PDF without footer-html and it is working correctly. Meaning I've added footer in the body content only. Generated PDF does not show any gap between that two sections.

**Comment by kavishatalsania** [comment]
I am able to solve this with `--footer-spacing -1`.

Please don't consider this workaround a fix. I would appreciate if this issue will remain opened until resolved.

**Comment by PhilterPaper** [comment]
Yeah, that sounds like a glitch in wkHTMLtoPDF, but at least you have a workaround. Just out of curiosity, does it do the same thing in wkHTMLtoImage?

**Comment by kavishatalsania** [comment]
I don't know. I have not checked for wkHTMLtoImage.

**Comment by dadazhouRenee** [comment]
same problem
What causes it

**Comment by 1pavanb** [comment]
yes @dadazhouRenee 

wkhtmltopdf 0.12.6 (with patched qt)

```
echo "<body style='background-color:black;padding=0;margin=0'>" | wkhtmltopdf -T 0 -B 0 -L 0 -R 0 --disable-smart-shrinking - bad.pdf
```
